workloads/generator.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+from enum import Enum
+import yaml
+import operator
+import functools
+import random
+import sys
+import json
+
+
+def finished(si, ni):
+    for i in si:
+        if i < ni - 1:
+            return False
+    return True
+
+
+workload_config_file = sys.argv[1]
+output_request_file = sys.argv[2]
+print('loading workload config from: ' + workload_config_file)
+
+with open(workload_config_file) as f:
+    config = f.read()
+
+data = yaml.load(config, Loader=yaml.Loader) # a list of dicts
+
+function_names = [f['name'] for f in data]
+arrival_rates = np.array([f['arrival_rate'] for f in data]) # num of invocations per second
+num_invocations = np.array([f['num_invocation'] for f in data])
+
+print('function names: ' + str(function_names))
+print('arrival rates: ' + str(arrival_rates))
+print('number of invocations: ' + str(num_invocations))
+
+
+mu = 1/arrival_rates
```

Comment:
hmm... I might be wrong but the documentation (which is really poorly worded...) says:
>\beta is the scale parameter, which is the inverse of the rate parameter \lambda = 1/\beta.
which I believe means that the input to the function is the average inter-arrival time. If this is the case, we would want to take the inverse, i.e., `mu = 1/arrival_rates`.

Reply:
OK, so I compared the probability density function in the [numpy.random.exponential documentation](https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.random.exponential.html) with [Wikipedia, “Exponential distribution”](http://en.wikipedia.org/wiki/Exponential_distribution). They're using the same notations.
So `scale` is the `1/lambda`, where `lambda` is the arrival rate. So I believe taking the inverse is necessary.

Reply:
> where lambda is the arrival rate

I don't believe that's true. The mean of the distribution is `1/ lambda`, and the mean of the distribution should equal the average (mean) arrival rate, no?

Reply:
Oh, I'm confused because I misinterpreted `arrival_rates` as in the adjacent comment thread---what it _means_ is the number of times per second (a rate) that requests arrive on average. What I _thought_ it meant was the average delta in milliseconds between subsequent requests.

Reply:
yes. sorry I should have been more clear in my comments (or added comments to `generator.py` in general). My bad... But just to make sure we're on the same page, there are 2 parameters:
1. The rate parameter `\lambda`, which is the `arrival_rates` variable in the code. This is the **average number of requests per unit of time**
2. The scale parameter `\beta`, which is the `mu` variable in the code. This is the inverse of `\lambda`, and it represents the **average inter-arrival time between requests**. It is also the mean of the exponential distribution (thus I named the variable `mu` in the code).